x-pack/plugin/core/src/main/config/log4j2.properties

```diff
@@ -115,3 +130,8 @@ logger.samlxml_decrypt.name = org.opensaml.xmlsec.encryption.support.Decrypter
 logger.samlxml_decrypt.level = fatal
 logger.saml2_decrypt.name = org.opensaml.saml.saml2.encryption.Decrypter
 logger.saml2_decrypt.level = fatal
+
+logger.behavioral_analytics_logfile.name = org.elasticsearch.xpack.application.analytics.AnalyticsEventLogger
```

Comment:
This class doesn't seem to be anything other than a wrapper around a Logger. Can it be created/managed outside of log4j, so that a non-custom logger is used? We have long term plans for locking down log4j, and the less custom things needed (like a custom logger), the easier that work will be.

Reply:
Given the expected amount of log, we tried to keep those events into a separate file (_analytics.log).

It seems to me that the easiest way to do it is to create a new logger that does not inherits parent loggers config (`additivity: false`). 

The audit log does more or less the same:

```
logger.xpack_security_audit_logfile.name = org.elasticsearch.xpack.security.audit.logfile.LoggingAuditTrail
logger.xpack_security_audit_logfile.level = info
logger.xpack_security_audit_logfile.appenderRef.audit_rolling.ref = audit_rolling
logger.xpack_security_audit_logfile.additivity = false
```

The biggest difference is that we format completely the event in our code.

What alternative implementation can you propose?

Reply:
@afoucret I think what you have would work, but this means that you have to maintain the schema inside `AnalyticsEvent` so that it always is ECS json

what I would suggest is to rely on our `ECSJsonLayout` - the same as other ES appenders. It should take care of emitting json log lines.
your AnalyticsEvent would no longer need to implement `ToXContentObject` 
you can then use Logger directly like this:
```
  logger.info(
                new ESLogMessage("some message").field("some field", "some value")
```

unless you expect this event to grow and contain a lot of fields?

Reply:
just a not so that we don't forget. We would need to update filebeat configs in cloud/onprem repos

Reply:
> We have long term plans for locking down log4j, and the less custom things needed (like a custom logger)

@rjernst bear in mind this is just a logger name `org.elasticsearch.xpack.application.analytics.AnalyticsEventLogger`. That class internally is justing a regular log4j2 logger. I think this should not be a problem to use es-logging in the future here.